src/edu/iastate/metnet/metaomgraph/chart/HeatMapChart.java

```diff
@@ -0,0 +1,340 @@
+/**
+ * 
+ */
+package edu.iastate.metnet.metaomgraph.chart;
+
+import edu.iastate.metnet.metaomgraph.AnimatedSwingWorker;
+import edu.iastate.metnet.metaomgraph.FrameModel;
+import edu.iastate.metnet.metaomgraph.IconTheme;
+import edu.iastate.metnet.metaomgraph.MetaOmGraph;
+import edu.iastate.metnet.metaomgraph.ui.TaskbarInternalFrame;
+import edu.iastate.metnet.metaomgraph.utils.Utils;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.Font;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFileChooser;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import org.jcolorbrewer.ColorBrewer;
+import org.jcolorbrewer.ui.ColorPaletteChooserDialog;
+
+/**
+ * @author sumanth
+ * HeatMapChart class is frame which holds the heatmap panel and button panels and all
+ * the user interaction related functions
+ */
+public class HeatMapChart extends TaskbarInternalFrame implements ActionListener{
+	
+	private double[][] originalData;
+	private double[][] heatMapData;
+	private String[] rowNames;
+	private String[] columnNames;
+	private String[] selectedDataCols;
+	
+	private JButton properties;
+	private JButton save;
+	private JButton changePalette;
+	private JButton splitDataset;
+	private JScrollPane scrollPane;
+	private JButton bottomPanelButton;
+	
+	private double minValue;
+	private boolean transposeData;
+	private boolean sampleDataOnRow;
+	private Map<String, Collection<Integer>> splitIndex;
+	private HeatMapPanel heatMap;
+	private Font textFont;
+	private Color textColor;
+	
+	/**
+	 * Constructor
+	 * @param data 2d heatmap data
+	 * @param rowNames row labels
+	 * @param columnNames column labels
+	 * @param transposeData transpose rows and columns?
+	 */
+	public HeatMapChart(double[][] data, String[] rowNames, String[] columnNames, boolean transposeData) {
```

Comment:
explain parameters e.g. data dimensions, whar are rowNames and colNames

Reply:
I think I mentioned the data dimensions as 2d data already.

Reply:
I meant to describe what are rows and columns. I think rows are features and columns are samples?